rfcs/0000-top-level-assumptions.md

```diff
@@ -0,0 +1,330 @@
+| `superIsCallableConstructor` | When this option is enabled, `super(arg1)` will be transpiled to `BaseClass.call(this, arg1)`. This means that it won't work with native classes or with built-ins, but only with compiled classes or ES5 constructors | `loose` in `transform-classes` | |
+| `constantSuper` | The `super` binding in classes can be changed using `setPrototypeOf`, so it's not possible to statically know it. With this option Babel can assume that the superclass is never changed at runtime. | `loose` in `transform-classes`, `proposal-class-properties`, `proposal-private-methods` and `proposal-decorators`. It should also be added to `transform-object-super`. | [🔗](https://babeljs.io/repl#?browsers=chrome%2040&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEBC0G8BQ1oFsCmAXAFgewBMAKASkWjBAwCcsiAiWAQnpIG5oBfJbpUSGAGFEKdNnzEyCClVoNBLdlx5I-4KNACC0DAA8sGAHYEY8ZKmoBXQ6RGpUAegfQAKgAkAkgGVoAGQ8AcgCiTKKoEJYADjQAdJi4hKRsoty8APIARgBWGMBYMRDYAArUeFhlAJ7RaQBmRJoxkaXlWFUYADTQgo3NldHsqoYYAO5apDFWNuxAA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env&prettier=false&targets=&version=7.10.1&externalPlugins=) |
+| `noDocumentAll` | When compiling the `??` and `?.` operators, assume that they are never used with `document.all` and thus `== null` is safe | `loose` in `proposal-optional-chaining` and `proposal-nullish-coalescing-operator` | [🔗](https://babeljs.io/repl#?browsers=chrome%2070&build=&builtIns=false&spec=false&loose=false&code_lz=IYfgdARg3AUMAEIT2kA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Cenv&prettier=false&targets=&version=7.10.1&externalPlugins=) | 3 |
+| `pureGetters` | When an expression that might invoke getters needs to be evaluated multiple times, Babel caches its value. When this option is enabled, the expression can be safely re-evaluated. For example, `a.b?.()` can be compiled to `a.b != null && a.b()` without caching `a.b` | `loose` in `proposal-optional-chaining` and in `proposal-object-rest-spread` _(not documented)_ | [🔗](https://babeljs.io/repl#?browsers=chrome%2070&build=&builtIns=false&spec=false&loose=true&code_lz=IYOgRg_CAUCUQ&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env&prettier=false&targets=&version=7.10.1&externalPlugins=) |
+| `setModuleMeta` | When compiling ESM to CJS, Babel defines a non-enumerable, non-witable, non-configurable `__esModule` property on the `exports` object. When this option is enabled, that property is set using a simple assignment | `loose` in `transform-modules-commonjs` |
```

Comment:
I was implementing this option, and what do you think about renaming it to `noModuleObjectIteration`?

The difference when this assumption is set, is that `__esModule` is enumerable so it will appear when iterating over a module object.

Reply:
I personally prefer `enumerableModuleMeta`.